### PR TITLE
centralize observe channel listener handling

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

To ensure that the observe channel handling
doesn't accidentally spawn unkillable routines,
adds a centralized channel listener func for
standard channel management in observe
progress bars.

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :bug: Bugfix

## Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test

### CODE REVIEW

**Comment by ntolia** [comment]
@ryanfkeepers Please add a changelog entry. This is a high-impact change for CLI users.

**Comment by vkamra** [comment]
@ryanfkeepers - changes look good (pending CHANGELOG request from @ntolia )

What can we do in terms of a regression test here. Is there a modification to `TestObserveProgressWithChannelClosed` that would have caught this?

File: src/internal/observe/observe.go [hunk 1]
```diff
@@ -469,7 +431,26 @@ func waitAndCloseBar(bar *mpb.Bar, log func()) func() {
 // other funcs
 // ---------------------------------------------------------------------------
 
-const Bullet = "∙"
+var nop = func() {}
+
+// listen runs a
```
Comment (vkamra) [inline src/internal/observe/observe.go:436]: Update comment

File: src/internal/observe/observe.go [hunk 2]
```diff
@@ -469,7 +431,26 @@ func waitAndCloseBar(bar *mpb.Bar, log func()) func() {
...
+func listen(ctx context.Context, ci <-chan struct{}, onEnd, onInc func()) {
```
Comment (vkamra) [inline src/internal/observe/observe.go:437]: RE: my comment about a regression test. I think we should at minimum add a unit test for this. Kudos for pulling this logic out - makes it testable!

Comment (ryanfkeepers) [reply src/internal/observe/observe.go]: done, tests have been added

**Pull request merged**